fms2_io/include/array_utils_char.inc

```diff
@@ -23,79 +23,202 @@
 !> @{
 
 !> @brief Allocate character arrays using an input array of sizes.
-subroutine allocate_array_char_1d(buf, sizes)
+subroutine allocate_array_char_1d(buf, sizes, initialize)
 
   character(len=*), dimension(:), allocatable, intent(inout) :: buf !< Array that will be allocated.
   integer, dimension(1), intent(in) :: sizes !< Array of dimension sizes.
+  logical, intent(in), optional :: initialize !< Optional argument when true will initialize with a blank string.
+
+  logical :: init !< local variable for initialize
+  integer :: i, c !< for looping
+
+  init = .false.
+  if (present(initialize)) init = initialize
 
   if (allocated(buf)) then
     deallocate(buf)
   endif
   allocate(buf(sizes(1)))
+
+  if (init) then
+    do i = 1, sizes(1)
+      do c = 1, len(buf(i))
+        buf(i)(c:c) = " "
```

Comment:
Does this workaround work without looping though? Like if you just did buf = " " or even buf(i) = " "

Reply:
This is the only way to ensure every character in the string array is initialized to zero.  I tested both buf = " " and buf(i) = " " and those will not properly initialize all characters in all elements of the string array.